pkg/kotsadmsnapshot/backup.go

```diff
@@ -354,39 +386,64 @@ func CreateInstanceBackup(ctx context.Context, cluster *downstreamtypes.Downstre
 	// marshal apps sequences map
 	b, err := json.Marshal(appsSequences)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to marshal apps sequences")
+		return "", errors.Wrap(err, "failed to marshal apps sequences")
 	}
 	marshalledAppsSequences := string(b)
 
 	// marshal apps versions map
 	b, err = json.Marshal(appVersions)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to marshal apps versions")
+		return "", errors.Wrap(err, "failed to marshal apps versions")
 	}
 	marshalledAppVersions := string(b)
 
-	// add kots annotations
-	backupAnnotations["kots.io/snapshot-trigger"] = snapshotTrigger
-	backupAnnotations["kots.io/snapshot-requested"] = time.Now().UTC().Format(time.RFC3339)
-	backupAnnotations["kots.io/instance"] = "true"
-	backupAnnotations["kots.io/kotsadm-image"] = kotsadmImage
-	backupAnnotations["kots.io/kotsadm-deploy-namespace"] = kotsadmNamespace
-	backupAnnotations["kots.io/apps-sequences"] = marshalledAppsSequences
-	backupAnnotations["kots.io/apps-versions"] = marshalledAppVersions
-	backupAnnotations["kots.io/is-airgap"] = strconv.FormatBool(kotsadm.IsAirgap())
+	now := time.Now()
+	backupName := fmt.Sprintf("backup-%d", now.UnixNano())
+
+	addCommonAnnotations := func(annotations map[string]string) map[string]string {
+		if annotations == nil {
+			annotations = make(map[string]string, 0)
+		}
+		annotations["kots.io/snapshot-trigger"] = snapshotTrigger
+		annotations["kots.io/snapshot-requested"] = now.UTC().Format(time.RFC3339)
+		annotations["kots.io/instance"] = "true"
+		annotations["kots.io/kotsadm-image"] = kotsadmImage
+		annotations["kots.io/kotsadm-deploy-namespace"] = kotsadmNamespace
+		annotations["kots.io/apps-sequences"] = marshalledAppsSequences
+		annotations["kots.io/apps-versions"] = marshalledAppVersions
+		annotations["kots.io/is-airgap"] = strconv.FormatBool(kotsadm.IsAirgap())
```

Comment:
should any or all of these be added to the app backup?

Reply:
wouldn't hurt to have them all